core/src/main/java/com/netease/arctic/data/DefaultKeyedFile.java

```diff
@@ -41,13 +41,31 @@ public class DefaultKeyedFile implements PrimaryKeyedFile, Serializable {
   private final DataFile internalFile;
 
   private transient FileMeta meta;
+  private transient ChangedLsn minLsn;
+  private transient ChangedLsn maxLsn;
 
   public DefaultKeyedFile(DataFile internalFile) {
     this.internalFile = internalFile;
   }
 
   private void parse() {
-    this.meta = FileUtil.parseFileMetaFromFileName(FileUtil.getFileName(internalFile.path().toString()));
+    this.meta = TableFileUtils.parseFileMetaFromFileName(TableFileUtils.getFileName(internalFile.path().toString()));
+    if (internalFile.lowerBounds() != null) {
+      ByteBuffer minOffsetBuffer = internalFile.lowerBounds().get(MetadataColumns.FILE_OFFSET_FILED_ID);
+      if (minOffsetBuffer != null) {
+        minLsn = ChangedLsn.of(meta.transactionId(), Conversions.fromByteBuffer(Types.LongType.get(), minOffsetBuffer));
+      } else {
+        minLsn = ChangedLsn.of(meta.transactionId(), Long.MAX_VALUE);
+      }
+    }
+    if (internalFile.upperBounds() != null) {
+      ByteBuffer maxOffsetBuffer = internalFile.upperBounds().get(MetadataColumns.FILE_OFFSET_FILED_ID);
+      if (maxOffsetBuffer != null) {
+        maxLsn = ChangedLsn.of(meta.transactionId(), Conversions.fromByteBuffer(Types.LongType.get(), maxOffsetBuffer));
+      } else {
+        maxLsn = ChangedLsn.of(meta.transactionId(), Long.MAX_VALUE);
+      }
+    }
```

Comment:
Maybe we could extract the common logic into a private method, for example:

```
private ChangedLsn parseChangedLsn(Map<Integer, ByteBuffer> map, FileMeta meta) {
    ByteBuffer offsetBuffer = map.get(MetadataColumns.FILE_OFFSET_FILED_ID);
      if (maxOffsetBuffer != null) {
        lsn = ChangedLsn.of(meta.transactionId(), Conversions.fromByteBuffer(Types.LongType.get(), offsetBuffer));
      } else {
        lsn = ChangedLsn.of(meta.transactionId(), Long.MAX_VALUE);
      }
      return lsn;
}
```

Reply:
this is original code, you could arise this in other PRs